src/CommunityCommons/javasource/communitycommons/StringUtils.java

```diff
@@ -297,21 +297,23 @@ public static String randomStrongPassword(int minLen, int maxLen, int noOfCAPSAl
 		if (minLen > maxLen) {
 			throw new IllegalArgumentException("Min. Length > Max. Length!");
 		}
-		if ((noOfCAPSAlpha + noOfDigits + noOfSplChars) > minLen) {
-			throw new IllegalArgumentException("Min. Length should be atleast sum of (CAPS, DIGITS, SPL CHARS) Length!");
+		if ((noOfCAPSAlpha + noOfDigits + noOfSplChars + 1) > minLen) {

```

Comment:
We would like to avoid new cases where this code can throw. Without the `+ 1` we will not include lower case characters if `noOfCAPSAlpha + noOfDigits + noOfSplChars == minLen` but I think it is reasonable the caller should pass a greater value of `minLen` if lower case letters are required.

We might want to make this a bit more explicit in documentation.

Reply:
If you can think of a better way to do this, please do. Meanwhile the bug remains. It would be nice to have this fixed.